### ISSUE TITLE
[BUG]: sonic_prefix_lists falsely claim changes are done/necessary if 'le' is 32 (IPv4) or 128 (IPv6)

### ISSUE THREAD

**Issue by toreanderson** [ansible-collections/dellemc.enterprise_sonic#361]
### Bug Description

When applying the playbook containing prefix lists with `le 32` (IPv4) or `le 128` (IPv6), each and every subsequent run will claim there are changes to the prefix lists, even though that is not the case. This happens in with and without `--check`. Furthermore,  `--diff` does not reveal what the claimed changes are, though the JSON output from `--verbose` gives a clue.

### Product Name

SONiC-OS-4.2.0-Enterprise_Base

### Component or Module Name

sonic_prefix_lists

### DellEMC Enterprise SONiC Ansible Collection Version

dellemc.enterprise_sonic 2.4.0

### SONiC Software Version

SONiC-OS-4.2.0-Enterprise_Base

### Configuration

```
CONFIG_FILE() = /home/debian/ansible/ansible.cfg
DEFAULT_HASH_BEHAVIOUR(/home/debian/ansible/ansible.cfg) = merge
DEFAULT_HOST_LIST(/home/debian/ansible/ansible.cfg) = ['/home/debian/ansible/hosts.yml']
DEFAULT_JINJA2_EXTENSIONS(/home/debian/ansible/ansible.cfg) = jinja2.ext.do
HOST_KEY_CHECKING(/home/debian/ansible/ansible.cfg) = False
INTERPRETER_PYTHON(/home/debian/ansible/ansible.cfg) = auto_silent
MAX_FILE_SIZE_FOR_DIFF(/home/debian/ansible/ansible.cfg) = 1048576
PERSISTENT_COMMAND_TIMEOUT(/home/debian/ansible/ansible.cfg) = 3000
```

### Steps to Reproduce

Run the following tasks at least twice:

```yaml
    - name: Configure prefix lists
      dellemc.enterprise_sonic.sonic_prefix_lists:
        config:
          - name: ROUTESERVERS-IN-IPV4
            afi: ipv4
            prefixes:
              - sequence: 65535
                action: deny
                prefix: 0.0.0.0/0
                le: 32
          - name: ROUTESERVERS-IN-IPV6
            afi: ipv6
            prefixes:
              - sequence: 65535
                action: deny
                prefix: ::/0
                le: 128
          - name: ROUTESERVERS-OUT-IPV4
            afi: ipv4
            prefixes:
              - sequence: 65535
                action: deny
                prefix: 0.0.0.0/0
                le: 32
          - name: ROUTESERVERS-OUT-IPV6
            afi: ipv6
            prefixes:
              - sequence: 65535
                action: deny
                prefix: ::/0
                le: 128
```

### Expected Behavior

a. Ansible-playbook should report `changed` on the initial run, when the prefix lists are created in the first place.
b. Ansible-playbook should report `ok` (that is: not `changed`) on subsequent runs.
c. Ansible-playbook should show a diff when run with `--diff` when it reports `changed`.

### Actual Behavior

a. Ansible-playbook repors `changed` on the initial run, when the prefix lists are created in the first place. (this is OK)
b. Ansible-playbook reports `changed` on subsequent runs.
c. Ansible-playbook does not show a diff when run with `--diff` when it reports `changed`.

### Logs

Subsequent runs show the following:

```
PLAY [ent_sonic] ***************************************************************

TASK [Configure prefix lists] **************************************************
changed: [leaf1]

PLAY RECAP *********************************************************************
leaf1   : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

### Screenshots

When running with `--verbose`,  the additional output reveals more of what is going on:

```
changed: [leaf1] => {"after": [{"afi": "ipv4", "name": "ROUTESERVERS-IN-IPV4", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "0.0.0.0/0", "sequence": 65535}]}, {"afi": "ipv6", "name": "ROUTESERVERS-IN-IPV6", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "::/0", "sequence": 65535}]}, {"afi": "ipv4", "name": "ROUTESERVERS-OUT-IPV4", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "0.0.0.0/0", "sequence": 65535}]}, {"afi": "ipv6", "name": "ROUTESERVERS-OUT-IPV6", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "::/0", "sequence": 65535}]}], "before": [{"afi": "ipv4", "name": "ROUTESERVERS-IN-IPV4", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "0.0.0.0/0", "sequence": 65535}]}, {"afi": "ipv6", "name": "ROUTESERVERS-IN-IPV6", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "::/0", "sequence": 65535}]}, {"afi": "ipv4", "name": "ROUTESERVERS-OUT-IPV4", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "0.0.0.0/0", "sequence": 65535}]}, {"afi": "ipv6", "name": "ROUTESERVERS-OUT-IPV6", "prefixes": [{"action": "deny", "ge": null, "le": null, "prefix": "::/0", "sequence": 65535}]}], "changed": true, "commands": [{"afi": "ipv4", "name": "ROUTESERVERS-IN-IPV4", "prefixes": [{"action": "deny", "le": 32, "prefix": "0.0.0.0/0", "sequence": 65535}], "state": "merged"}, {"afi": "ipv6", "name": "ROUTESERVERS-IN-IPV6", "prefixes": [{"action": "deny", "le": 128, "prefix": "::/0", "sequence": 65535}], "state": "merged"}, {"afi": "ipv4", "name": "ROUTESERVERS-OUT-IPV4", "prefixes": [{"action": "deny", "le": 32, "prefix": "0.0.0.0/0", "sequence": 65535}], "state": "merged"}, {"afi": "ipv6", "name": "ROUTESERVERS-OUT-IPV6", "prefixes": [{"action": "deny", "le": 128, "prefix": "::/0", "sequence": 65535}], "state": "merged"}]}
```

If I copy'n'paste this JSON blob into a file, I observe the following clues:

1. The `before` and `after` blobs are exactly the same. This probably explains the missing output from `--diff`.
2. Both the `before` and `after` blobs contain `"le": null`. This is obviously not correct, and probably explains why the module feels the need to change something.
3. Diffing the `before`/`after` blobs with `commands` reveals an attempted change to the `le` attribute:

```diff
$ diff -u <(jq .before < out.json ) <(jq .commands < out.json)
--- /dev/fd/63  2024-03-26 11:26:07.403271295 +0100
+++ /dev/fd/62  2024-03-26 11:26:07.404271298 +0100
@@ -5,12 +5,12 @@
     "prefixes": [
       {
         "action": "deny",
-        "ge": null,
-        "le": null,
+        "le": 32,
         "prefix": "0.0.0.0/0",
         "sequence": 65535
       }
-    ]
+    ],
+    "state": "merged"
   },
   {
     "afi": "ipv6",
@@ -18,12 +18,12 @@
     "prefixes": [
       {
         "action": "deny",
-        "ge": null,
-        "le": null,
+        "le": 128,
         "prefix": "::/0",
         "sequence": 65535
       }
-    ]
+    ],
+    "state": "merged"
   },
   {
     "afi": "ipv4",
@@ -31,12 +31,12 @@
     "prefixes": [
       {
         "action": "deny",
-        "ge": null,
-        "le": null,
+        "le": 32,
         "prefix": "0.0.0.0/0",
         "sequence": 65535
       }
-    ]
+    ],
+    "state": "merged"
   },
   {
     "afi": "ipv6",
@@ -44,11 +44,11 @@
     "prefixes": [
       {
         "action": "deny",
-        "ge": null,
-        "le": null,
+        "le": 128,
         "prefix": "::/0",
         "sequence": 65535
       }
-    ]
+    ],
+    "state": "merged"
   }
 ]
```

Through experimentation I have found that this only happens for `le` values 32 and 128. If I change `le` to 31 and 127, subsequent playbook runs report `ok` and the `commands` blob in the JSON contains an empty array. The same happens if I remove `le` altogether. (Obviously this changes which prefixes are actually matched by the prefix list, so it cannot be used as a workaround.)

Finally, I'd like to note that specifying `le 32/128` actually works as expected, in the sense that they end up being configured as such:

```
leaf1# show running-configuration | grep ROUTESERVERS
ip prefix-list ROUTESERVERS-IN-IPV4 seq 65535 deny 0.0.0.0/0 le 32
ip prefix-list ROUTESERVERS-OUT-IPV4 seq 65535 deny 0.0.0.0/0 le 32
ipv6 prefix-list ROUTESERVERS-IN-IPV6 seq 65535 deny ::/0 le 128
ipv6 prefix-list ROUTESERVERS-OUT-IPV6 seq 65535 deny ::/0 le 128
```

(Note: I put the above in the «Screenshots» text area, since the «Additional Information» only accepts a single line.)

**Comment by ArunSaravananBalachandran** [comment]
This issue has been resolved by #354

**Comment by kerry-meyer** [comment]
The submitter for the fix to this issue posted an update after completing the fix. Because the submitter has indicated approval of the fix, I am closing this issue.
